src/concurrency/sync.rs

```diff
@@ -121,8 +121,10 @@ declare_id!(CondvarId);
 struct CondvarWaiter {
     /// The thread that is waiting on this variable.
     thread: ThreadId,
-    /// The mutex on which the thread is waiting.
-    mutex: MutexId,
+    /// The mutex or rwlock on which the thread is waiting.
+    lock: u32,
```

Comment:
Not sure if that's overkill, but it may be a bit cleaner to use a

```rust
enum Lock {
    Mutexes(MutexId),
    RwLock{ id: RwLockId, shared: bool }
}
```

Reply:
I'd actually like to go further and replace `shared: bool` by an `enum RwLockMode { Read, Write }` or something like that.